src/lib.rs

```diff
@@ -225,18 +227,30 @@ cfg_if! {
                   target_env = "sgx",
               )))] {
         #[path = "rdrand.rs"] mod imp;
-    } else if #[cfg(target_arch = "wasm32")] {
+    // ideally we would like to use `target = "wasm32-unknown-unknown"`,
```

Comment:
Personally I find something like:
```rust
...
} else if #[cfg(all(target_arch = "wasm32", feature = "wasm-bindgen"))] {
    #[path = "wasm32_bindgen.rs"] mod imp;
} else if #[cfg(all(target_arch = "wasm32", feature = "stdweb"))] {
    #[path = "wasm32_stdweb.rs"] mod imp;
} else if #[cfg(feature = "dummy")] {
    #[path = "dummy.rs"] mod imp;
} else {
    compile_error!("\
...
```
to be a little more readable. It avoids nested `cfg_if!`s, and we don't need the `unknown` specifier. If we get here, the other `wasm32` cases are taken care of.

In theory, a new `wasm32` target could be added that's not addressed in the above, but having the `stdweb`/`wasm-bindgen` features work in that case seems fine (even ideal).

Reply:
I am not sure about not needing the `unknown` specifier, imagine that we don't have WASI or Emscripten, with this config would not be supported for such "new" targets. But I guess since we can't use `target_vendor` here, we are not completely future proof anyway, so it makes sense to simplify the code.